router.go

```diff
@@ -158,6 +163,11 @@ func (m MapRouter) Choose(ctx context.Context, req *transport.Request) (transpor
 		service = m.defaultService
 	}
 
+	if _, ok := m.allServices[service]; !ok {
+		fmt.Println(m.allServices)
+		return transport.HandlerSpec{}, yarpcerrors.Newf(yarpcerrors.CodeUnimplemented, "unrecognized service name %q", req.Service)
```

Comment:
optional: should we provide the list of service names that are supported?

Reply:
Thanks for pointing out! fmt.Println statement for debugging has been removed.

Reply:
I didn't mean the `fmt.Println` (although it's good you removed that).

I meant that if the service is one that we don't recognize, should we report a more friendly error message that lists the service names that we *do* recognize.

Reply:
Sorry, I misunderstand your suggestion at the beginning. Now the error message will contain all available service names.